plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/DirectVifDriver.java

```diff
@@ -47,12 +47,7 @@
     public LibvirtVMDef.InterfaceDef plug(NicTO nic, String guestOsType, String nicAdapter, Map<String, String> extraConfig) throws InternalErrorException, LibvirtException {
         LibvirtVMDef.InterfaceDef intf = new LibvirtVMDef.InterfaceDef();
 
-        if (nic.getType() == Networks.TrafficType.Guest) {
-            Integer networkRateKBps = (nic.getNetworkRateMbps() != null && nic.getNetworkRateMbps().intValue() != -1) ? nic.getNetworkRateMbps().intValue() * 128 : 0;
-            intf.defDirectNet(_libvirtComputingResource.getNetworkDirectDevice(), null, nic.getMac(), getGuestNicModel(guestOsType, nicAdapter),
-                _libvirtComputingResource.getNetworkDirectSourceMode(), networkRateKBps);
-
-        } else if (nic.getType() == Networks.TrafficType.Public) {
+        if (nic.getType() == Networks.TrafficType.Guest || nic.getType() == Networks.TrafficType.Public) {
```

Comment:
We could create a static `Set`, add the types `Networks.TrafficType.Guest` and `Networks.TrafficType.Public`, and then verify if the set contains the `nic.getType()`. This way we can avoid long statements, as, if we need add another type, we just have to add it in the `Set`.

Reply:
I would considder
```suggestion
        if (Sets.newHashSet(Networks.TrafficType.Guest,
                            Networks.TrafficType.Public).contains(nic.getType()) {
```
but I don't think it is an improvement with only two elements to check. Like the pattern for larger sets, but At only two elements it doesn't make it more readable, certainily not with the separate construction above the condition.